spiketoolkit/postprocessing/features.py

```diff
@@ -0,0 +1,143 @@
+"""
+Uses the functions in SpikeInterface/spikefeatures to compute
+unit template features
+"""
+
+import pandas as pd
+from spikefeatures import features
+from scipy.signal import resample
+from .postprocessing_tools import get_unit_templates, get_unit_max_channels
+
+
+def compute_unit_template_features(
+        sorting,
+        recording,
+        unit_ids=None,
+        feature_names=None,
+        save_property_or_features=False,
+        recovery_slope_window=0.7,
+        upsampling_factor=1,
+        invert_waveforms=False,
+        spread_threshold=None,
+        site_range=None,
+        site_spacing=None,
+        epoch_name=None,
+        channel_map=None,
+        return_dict=False,
+        verbose=False,
+
+):
+    """
+    Use SpikeInterface/spikefeatures to compute features for the unit
+    template. These consist of a set of 1D features:
+    - peak to valley (peak_to_valley), time between peak and valley
+    - halfwidth (halfwidth), width of peak at half its amplitude
+    - peak trough ratio (peak_trough_ratio), amplitude of peak over amplitude of trough
+    - repolarization slope (repolarization_slope), slope between trough and return to base
+    - recovery slope (recovery_slope), slope after peak towards baseline
+
+    And 2D features:
+
+    The metrics are computed on 'negative' waveforms, if templates are saved as
+    positive, pass keyword 'invert_waveforms'.
+
+    Recording should be bandpassfiltered  # TODO: make this a requirement?
+
+    Parameters:
+    -------
+    sorting: SortingExtractor
+        The sorting result to be evaluated.
+    recording: RecordingExtractor
+        The given recording extractor from which to extract amplitudes
+    unit_ids: list
+        List of unit ids to compute metric for. If not specified, all units are used
+    return_dict: bool
+        If True, will return dict of metrics (default False)
+    feature_names: list
+        list of feature names to be computed
+    upsampling_factor : int
+        factor with which to upsample the template resolution (default 1)
+
+    TODO:
+    save_property_or_features=False,
+    recovery_slope_window=0.7,
+    invert_waveforms=False,
+    spread_threshold=None,
+    site_range=None,
+    site_spacing=None,
+    epoch_name=None,
```

Comment:
epochs have been finally removed from quality metrics

Reply:
removed

Reply:
and now also from features!